CMakeLists.txt

```diff
@@ -1,11 +1,69 @@
-cmake_minimum_required(VERSION 3.5.1)
+cmake_minimum_required(VERSION 3.7)
 
 find_program(CCACHE_PROGRAM ccache)
 if(CCACHE_PROGRAM)
   set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
   message(STATUS "ccache enabled (${CCACHE_PROGRAM})")
 endif()
 
+##################################
+## setup Hunter and its modules ##
+##################################
+SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
+SET(HUNTER_STATUS_PRINT ON  CACHE BOOL "Verbose info")
+SET(HUNTER_STATUS_DEBUG OFF CACHE BOOL "Debug info"  )
+
+# hunter binary cache server
+set(
+    HUNTER_CACHE_SERVERS
+    "https://github.com/soramitsu/hunter-binary-cache;https://github.com/elucideye/hunter-cache;https://github.com/ingenue/hunter-cache"
+    CACHE
+    STRING
+    "Hunter cache servers"
+)
+
+# enable this option to upload your binary cache
+# GITHUB_USER=<github_user> environment variable must be set
+# GITHUB_USER_PASSWORD=<github_personal_token> environment variable must be set
+option(HUNTER_RUN_UPLOAD "Upload cache binaries" OFF)
+if(HUNTER_RUN_UPLOAD)
+  if(NOT DEFINED "ENV{GITHUB_USER}")
+    message(FATAL_ERROR "HUNTER_RUN_UPLOAD=ON, but GITHUB_USER env variable is empty")
+  endif()
+  if(NOT DEFINED "ENV{GITHUB_USER_PASSWORD}")
+    message(FATAL_ERROR "HUNTER_RUN_UPLOAD=ON, but GITHUB_USER_PASSWORD env variable is empty, set to personal access token from github")
+  endif()
+endif()
+
+
+set(
+    HUNTER_PASSWORDS_PATH
+    "${CMAKE_SOURCE_DIR}/cmake/hunter-binary-cache.cmake"
+    CACHE
+    FILEPATH
+    "Hunter passwords"
+)
+
+#########################
+## configuration types ##
+#########################
+set(CMAKE_CONFIGURATION_TYPES "Debug;Release;Coverage" CACHE STRING "" FORCE)
```

Comment:
this is not needed, will be introduced in https://github.com/hyperledger/iroha/pull/1055

Reply:
We do not plan to merge 1055 now. Please do not confuse reviewers and the author